### DISCUSSION TITLE
NoSQL/SQL ORM - Prior Work

### DISCUSSION THREAD

**Discussion by ALPSMAC** [tokio-rs/toasty#52] [Ideas]
In your announcement post ( https://tokio.rs/blog/2024-10-23-announcing-toasty ) you asked for examples of prior work targeting both SQL and NoSQL databases.  

I wanted to call to your attention Quill from the Scala ecosystem if you're not yet familiar: https://zio.dev/zio-quill/contexts/ 

There are some unique ideas in there about using compile-time macros / meta-programming to support a variety of different databases and dialects, including at least one "NoSQL" example - Cassandra. 

Kind Regards,
Andy

**Comment by carllerche** [comment]
Thank you for the link. I'll check it out. Have you used it before? Do you have specific thoughts on what it does well and what it doesn't do as well?
